drivers/ethernet/eth_lan865x.c

```diff
@@ -373,16 +373,32 @@ static void lan865x_read_chunks(const struct device *dev)
 	k_sem_take(&ctx->tx_rx_sem, K_FOREVER);
 	pkt = net_pkt_rx_alloc(K_MSEC(cfg->timeout));
 	if (!pkt) {
-		LOG_ERR("OA RX: Could not allocate packet!");
+		/*
+		 * Disable RX channel as host temporary runs out of
+		 * available memory for allocating packets.
+		 */
+		oa_tc6_reg_rmw(tc6, LAN865x_MAC_NCR,
+			       LAN865x_MAC_NCR_RXEN, 0);
+
 		k_sem_give(&ctx->tx_rx_sem);
+		/*
+		 * Allow only TX, so net stack have time to process
+		 * already received data.
+		 */
+		k_msleep(cfg->timeout / 10);
```

Comment:
I am not a big fan of adding sleeps in the code, sometimes they work but it is difficult to decide how long to sleep etc. 
Would it be possible to change this so that the TX side informs when the RX side can proceed, perhaps via a mutex with a timeout?

Reply:
@jukkar - the time here is related to the timeout used previously with RX buffer allocation (`cfg->timeout`). The assumption here is that after 1/10 of time which we waited for RX packet to be allocated there shall be some memory available (as some packets are "consumed" by network stack).

Maybe, I shall change the commit message of this particular commit (as it seems to be misleading)? The problem which I've been solving is when I do try to send too much data in packets from HOST to LAN8651. As we do have some fixed number of available memory for RX chunks (rather smaller number) the "stall" caused by no memory in RX pool happens from time to time. I could "rely" on LAN8651 HW feature as it overwrites the buffer, but presented here solution seems to be simpler and less intrusive.

Reply:
I've discarded this commit as it brought more issues (with the sleep) than benefits. The driver shall behave in a robust way without it.